archai/nlp/nvidia_transformer_xl/mem_transformer.py

```diff
@@ -841,8 +849,8 @@ def _forward(self, dec_inp, mems=None, past_key_values=None):
             for i, (layer, past_key_values_i) in enumerate(zip(self.layers, past_key_values)):
                 hids.append(core_out.detach())
                 mems_i = None if mems is None else mems[i]
-                core_out, past_key_values_i = layer(core_out, pos_emb, self.r_w_bias,
-                                                    self.r_r_bias, dec_attn_mask=dec_attn_mask,
+                core_out, past_key_values_i = layer(core_out, pos_emb, getattr(self, f'r_w_bias_{i}'),
```

Comment:
This is pretty odd. Let's email Mojan?

Reply:
E-mailed her. I will come back with her response.

Reply:
"I don't remember the exact reason but I think the parameters inside the list were not being counted as trainable parameters inside PyTorch. And that was causing issues when exporting to ONNX (The onnx graph was incomplete and didn't have these parameters inside if I remember correctly). So that's why I ditched the list wrapper and initialized each parameter individually. Sorry, I don't remember the exact errors that were happening, but I think we shouldn't go back to lists."